### PR TITLE
Content controller

### PR DESCRIPTION
**Pull request by ctlong**
Add controller and views for content.

### CODE REVIEW

File: app/assets/javascripts/contents.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (h8rry) [inline app/assets/javascripts/contents.js]: If you don't need this, please remove

File: app/assets/stylesheets/contents.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Contents controller here.
```
Comment (h8rry) [inline app/assets/stylesheets/contents.scss]: If you don't need this, please remove

File: app/views/contents/index.jbuilder [hunk 3]
```diff
@@ -0,0 +1,10 @@
+json.array! @contents do |content|
+  json.id @content.id
+  json.form_id @content.form_id
+  json.type @content.type
+  json.index @content.index
+  json.label @content.label
+  json.options @content.options
+  json.created_at @content.created_at
+  json.updated_at @content.updated_at
+end
```
Comment (h8rry) [inline app/views/contents/index.jbuilder]: New line at the end of the file please

File: app/views/contents/create.jbuilder [hunk 4]
```diff
@@ -0,0 +1,8 @@
+json.id @content.id
+json.form_id @content.form_id
+json.type @content.type
+json.index @content.index
+json.label @content.label
+json.options @content.options
+json.created_at @content.created_at
+json.updated_at @content.updated_at
```
Comment (h8rry) [inline app/views/contents/create.jbuilder]: New line at the end of the file please

File: app/views/contents/index.jbuilder [hunk 5]
```diff
@@ -0,0 +1,10 @@
+json.array! @contents do |content|
+  json.id @content.id
```
Comment (h8rry) [inline app/views/contents/index.jbuilder]: Is it `json.id @content.id` or `json.id content.id`? Have you tested this?

Comment (h8rry) [inline app/views/contents/index.jbuilder]: Be careful if you were copy and pasting

File: app/controllers/contents_controller.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+class ContentsController < ApplicationController
+  def index
+    @contents = Form.find(params[:id]).contents.includes(:submissions)
```
Comment (h8rry) [inline app/controllers/contents_controller.rb]: How does your endpoints look like? `forms/:form_id/contents`? 

If so, you should do

``` ruby
@contents = Form.find(params[:form_id]).contents.includes(:submissions)
```

File: app/controllers/contents_controller.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  def create
+    @content = Form.find(params[:id]).contents.create(content_params)
+  end
+
+  def update
+    @content = Content.find(params[:id])
+    if @content.nil?
+      render json: {message: 'Cannot find content'}, status: :not_found
+    else
+      @content.update(content_params)
```
Comment (h8rry) [inline app/controllers/contents_controller.rb]: Technically, I would even do a `if @content.update(content_params)` to make sure the program can handle the case when the update didn't work. Same thing for delete

**Comment by h8rry** [comment]
@ctlong Please review and modify. You should also rebase since we merged previously.

**Comment by ctlong** [comment]
@ggharry I have modified and rebased the this branch as per your suggestions.

File: app/controllers/contents_controller.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    end
+  end
+
+  def delete
+    @content = Content.find(params[:id])
+    if @content.nil?
+      render json: {message: 'Cannot find content'}, status: :not_found
+    elsif @content.destroy
+      render json: {deleted: true}, status: :success
+    end
```
Comment (h8rry) [inline app/controllers/contents_controller.rb:33]: More like this

``` ruby
    if @content.nil?
      render json: {message: 'Cannot find content'}, status: :not_found
    else 
        if @content.destroy
            render json: {deleted: true}, status: :not_content
        else
            render json: {deleted: false}, status: :bad_request
        end    
    end
```

**Comment by h8rry** [comment]
The rest looks good. You can merge after reviewing

**Pull request merged**